src/hip_memory.cpp

```diff
@@ -2164,8 +2181,12 @@ hipError_t hipIpcCloseMemHandle(void* devPtr) {
         hipStatus = hipErrorInvalidValue;
     } else {
 #if USE_IPC
-        hsa_status_t hsa_status = hsa_amd_ipc_memory_detach(devPtr);
-        if (hsa_status != HSA_STATUS_SUCCESS) return hipErrorInvalidResourceHandle;
+        am_status_t am_status = hc::am_memtracker_remove(devPtr);
+        if (am_status == AM_SUCCESS) {
+            hsa_status_t hsa_status = hsa_amd_ipc_memory_detach(devPtr);
+            if (hsa_status != HSA_STATUS_SUCCESS) 
+                hipStatus = hipErrorInvalidResourceHandle;
+        } else hipStatus = hipErrorInvalidValue;
```

Comment:
How about logging an error on am_memtracker_remove error return, assuming that it should not really happen but if it does it would be good to have the log?

Reply:
Added these checks.